modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotStorage.scala

```diff
@@ -0,0 +1,71 @@
+package org.tessellation.infrastructure.snapshot
+
+import cats.data.NonEmptyList
+import cats.effect.{Async, Ref}
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.option._
+import cats.syntax.order._
+import cats.syntax.traverse._
+
+import scala.util.control.NoStackTrace
+
+import org.tessellation.dag.snapshot.{GlobalSnapshot, SnapshotOrdinal, StateChannelSnapshotWrapper}
+import org.tessellation.domain.snapshot.GlobalSnapshotStorage
+import org.tessellation.ext.crypto._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.address._
+import org.tessellation.security.hash.Hash
+
+object GlobalSnapshotStorage {
+
+  def make[F[_]: Async: KryoSerializer: Ref.Make](genesis: GlobalSnapshot): F[GlobalSnapshotStorage[F]] =
+    Ref[F].of[NonEmptyList[GlobalSnapshot]](NonEmptyList.of(genesis)).map(ref => make(ref))
+
+  def make[F[_]: Async: KryoSerializer: Ref.Make](
+    snapshotsRef: Ref[F, NonEmptyList[GlobalSnapshot]]
```

Comment:
Then  how would you name this argument `snapshots.modify { argument => }`? If you would also use `snapshots`, then it would shadow the outer name and would be unambiguous when reading the code.